examples/methods/ada_svr.py

```diff
@@ -0,0 +1,191 @@
+
+# compare that the coefficiants are the same that the one of SVR
+assert np.max(np.abs(beta_hat - beta_hat_svr.T[:, 0])) < 1e-4
+
+#############################################################################
+# This coefficient of SVR is an estimation of the importance of variables.
+# For estimating the confidence interval of the importance of variables,
+# ADA-SVR uses propose to estimate the distribution of the coefficients
+# of SVR, instead of using the classical permutation test for this estimation.
+
+#############################################################################
+# Estimation of the distribution of the coefficients of SVR
+# ---------------------------------------------------------
+#
+
+proba = permutation_test(
+    X, y, estimator=estimator, n_permutations=10000, n_jobs=8, seed=42, proba=True
+)
+plot_compare_proba_estimated(proba, beta_hat, scale)
+
+#############################################################################
+# **Compare the distribution of the coefficients of SVR with the
+# estimation of the distribution of the coefficients by ADA-SVR**
+print(
+    "ADA-SVR assumes that the normal distribution of the coefficient is",
+    "center around zeros.\n",
+    "Our estimation is that the maximum deviation is: {:.4f}\n".format(
+        np.max(np.abs(np.mean(proba, axis=0)))
+    ),
+    "ADA-SVR provides that the standard deviation the normal distribution",
+    "for each coefficients.\n",
+    "The maximum of difference between AdA-SVR estimation and our estimation",
+    "is:{:.4f}".format(np.max(np.abs(scale - np.std(proba, axis=0)) / scale)),
+)
+
+#############################################################################
+# Assumptions, Advantages and Disadvantages
+# -----------------------------------------
+#
+# **Assumptions**:
+#
+# - The distribution of the coefficients of SVR is normal centred around zeros.
+# - The method is valid for large sample sizes.
+# - The method has the linear models assumptions: linearity, normality, 
+#   homoscedasticity, independence, fixed features, absence of multicollinearity
+#   (see the book of :footcite:ct:`Molnar 2012<molnar2020interpretable>`
+#   for details)
+#
+# **Advantages**:
+# 
+# - The method is fast because it uses the central limit theorem to estimate
+#   the standard deviation of the distribution of the coefficients of SVR.
+# - The method has the advantage of linear models: transparency of 
+#   the prediction, high level of collective experiance and expertise 
+#   and a guarantee of convergence. (see the book of 
+#   :footcite:ct:`Molnar 2012<molnar2020interpretable>` for details)
+#
+# **Disadvantages**:
+#
+# - The method assumes that the distribution of the coefficients of SVR is normal centred around zeros.
+# - The method is not valid for small sample sizes.
+# - The method has all the disadvantages of linear models: only for linear 
+#   relationships, not good predicting performance, unintuitive. 
```

Comment:
unintuitive:
Following the explanation of Molnar, the interpretation of the weight can be counter-intuitive when the dimension is higher than 4. (see the last paragraph of [linear model](https://christophm.github.io/interpretable-ml-book/limo.html#disadvantages) )

Reply:
So you means that the model is "non-sparse", making it harder to figure out.

Reply:
No, my understanding of the argument of the Molnar is that when the dimension is higher than 4, it's quite impossible to image the hyperplane provided by the weights.

From Molnar's book : "The interpretation of a weight **can be unintuitive** because it depends on all other features. A feature with high positive correlation with the outcome y and another feature might get a negative weight in the linear model, because, given the other correlated feature, it is negatively correlated with y in the high-dimensional space. Completely correlated features make it even impossible to find a unique solution for the linear equation. An example: You have a model to predict the value of a house and have features like number of rooms and size of the house. House size and number of rooms are highly correlated: the bigger a house is, the more rooms it has. If you take both features into a linear model, it might happen, that the size of the house is the better predictor and gets a large positive weight. The number of rooms might end up getting a negative weight, because, given that a house has the same size, increasing the number of rooms could make it less valuable or the linear equation becomes less stable, when the correlation is too strong."

Reply:
Sure, but there are 2 ways to consider the problem:
* if your data is high-dimensional (dimension > 4) you cannot reliably interpret the weights ---which is True or False , depending on the correlation between features. But then, interpretability is not related to the model, only to the data...
* if your model is sparse, i.e. can explain  y with very few features, it becomes interpretable again. A Lasso regression or a regression tree are "sparse" in that sense. However, this is not a good idea, because the sparse solution rached by the model may not be significantly better than other sparse solutions.

Overall, I think that this is a bad argument. In any case, it is not related to ada-svr.